core-bundle/src/Resources/public/core.js

```diff
@@ -992,10 +994,53 @@ var Backend =
 	},
 
 	/**
-	 * Get the current scroll offset and store it in a cookie
+	 * Store the current scroll offset in sessionStorage
 	 */
 	getScrollOffset: function() {
-		document.cookie = "BE_PAGE_OFFSET=" + window.getScroll().y + "; path=" + (Contao.path || '/');
+		window.sessionStorage.setItem('contao_backend_offset', parseInt(window.getScroll().y, 10));
+	},
+
+	/**
+	 * Remove the legacy BE_PAGE_OFFSET cookie, scroll to the current offset if
+	 * it was defined and add the "down" CSS class to the header.
+	 */
+	initScrollOffset: function() {
+		// Kill the legacy cookie here; this way it can be sent by the server
+		// but it wont be resent by the client in the next request
+		Cookie.dispose('BE_PAGE_OFFSET');
```

Comment:
Are you sure this works and the cookie is never "http-only"?

Reply:
It won't work if someone sets the cookie `http-only`. I think the use case is just the one where people copied the code to their own `DC_*` classes or so. I guess if they did, they just copied `System::setCookie('BE_PAGE_OFFSET', 0, 0);`. We could also just adjust `System::setCookie()` and make it ignore the `BE_PAGE_OFFSET` cookie, wdyt @leofeyer?

Reply:
Well of course it was not `http-only` so far. `Backend.getScrollOffset()` set the cookie so it was never `http-only`. It could've only been `http-conly` if you set it using `System::setCooke()` which - if you did so - would've lead to weird behaviour. So I guess this is good then.